sklearn/utils/validation.py

```diff
@@ -576,6 +641,12 @@ def check_array(
     estimator : str or estimator instance, default=None
         If passed, include the name of the estimator in warning messages.
 
+    data_name : str, default=""
```

Comment:
I prefer `input_name`, since a lot of the wording in `check_array`'s docstring refers to an "input".